### PR TITLE
Add support for web/desktop

### PR DESCRIPTION
**Pull request by elliotsayes**
Add full support for web + desktop. Also fix formatting of iOS modal, and add filtering of relevant wallets (copy of desktop implementation)

### CODE REVIEW

File: lib/src/components/modal_main_page.dart [hunk 1]
```diff
@@ -44,22 +45,14 @@ class _ModalMainPageState extends State<ModalMainPage> {
...
-                          () => _groupValue = 0,
-                        ),
-                      ),
-                      1: _Segment(
-                        text: 'QR Code',
-                        onTap: () => setState(
-                          () => _groupValue = 1,
-                        ),
-                      ),
+                      0: Utils.isDesktop() ? qrSegment() : listSegment(),
```
Comment (elliotsayes) [inline lib/src/components/modal_main_page.dart]: This part is a little bit hacky, but it allows the QR to be displayed first on desktop (imitating the JS implementation). Display order is determined by map order and not the key.

File: lib/src/components/modal_main_page.dart [hunk 2]
```diff
@@ -77,6 +70,14 @@ class _ModalMainPageState extends State<ModalMainPage> {
       ),
     );
   }
+
+  _Segment listSegment() => _Segment(
```
Comment (3ph) [inline lib/src/components/modal_main_page.dart]: @elliotsayes Can you please convert these to StatelessWidget which is preferable to functions? I know it's a bit of an overkill but rather keep it in line with best practices.

File: lib/src/components/modal_wallet_ios_page.dart [hunk 3]
```diff
@@ -108,4 +110,23 @@ class ModalWalletIOSPage extends StatelessWidget {
       },
     );
   }
+
+  Future<List<Wallet>> getIOSWallets() {
```
Comment (3ph) [inline lib/src/components/modal_wallet_ios_page.dart]: @elliotsayes Can you get rid of the `get` as suggested in Effective Dart?

File: lib/src/utils/utils.dart [hunk 4]
```diff
@@ -1,31 +1,57 @@
+import 'package:flutter/foundation.dart';
 import 'package:url_launcher/url_launcher.dart';
 
 import '../models/wallet.dart';
 
 class Utils {
+  static bool isIOS() => defaultTargetPlatform == TargetPlatform.iOS;
```
Comment (3ph) [inline lib/src/utils/utils.dart]: @elliotsayes Can you please turn these into properties so it's in line with `TargetPlatform` properties?

**Review by 3ph — CHANGES_REQUESTED** [review]
Thanks a lot for the PR. Seems ok, just a couple of stylistic changes.

File: lib/src/components/modal_main_page.dart (hunk 2 above)
Comment (elliotsayes) [reply lib/src/components/modal_main_page.dart]: I didn't realise this, thanks

**Comment by elliotsayes** [comment]
Hi, made the changes you suggested. Thanks for the tips :)

**Review by 3ph — APPROVED** [review]

**Pull request merged**